### PR TITLE
Colour scale autoscaling in SliceViewer and ColorFill plots

### LINKED ISSUE #14818: LiveViewing in ColorFill Plot and SliceViewer
Users want to set the colour scale on these tools, but every time the new data comes through the scales get re-written. Should be a way to permanently set the scales so that re-scaling doesn't happen upon workspace reload.

This sounds like a small annoying problem, but it's actually stopping users using the live data mode at ISIS

Ask Owen for more details.

### PR DESCRIPTION
**Pull request by tom-perkins**
Resolves #14818 

Linked to discussion in #14834 

When the SliceViewer or a Color Fill plot is being used to visualise live data, the user's manually set colour scale should not be reset every time the data is updated. This PR enables the autoscaling of the colour bar to be disabled.

_Color fill plots_
The global option under `View > Preferences > 2D Plots > Autoscale` is used to determine whether or not the colour scale will rescale automatically. 

_SliceViewer_
A checkbox is placed on the UI to control autoscaling. The checkbox takes its default state from the global config option as above.

**Test**
- Start a fake live data listener as [here](http://www.mantidproject.org/MBC_Live_Data_Simple_Examples), e.g. ISIS_Histogram.

**_1. Color fill plot**_
- Launch a colour fill plot from the workspace containing live data
- Change the colour scale and check whether it remains the same when new data arrives
- Repeat with the `View > Preferences > 2D Plots > Autoscale` option toggled. Verify that the behaviour is different.

**_2. SliceViewer**_
- Launch the SliceViewer from the workspace containing live data
- Check that the state of the "Autoscale" checkbox matches the global option
- Change the colour scale with "Autoscale" checked and verify the scale is reset when more data comes in
- Uncheck "Autoscale" and repeat. Verify that the scale is not reset.

**To do:**
- [x] Update release notes

### CODE REVIEW

File: MantidPlot/src/Graph.cpp [hunk 1]
```diff
@@ -968,6 +968,10 @@ void Graph::enableAutoscaling(bool yes) {
       d_plot->setAxisScaleDiv(i, *d_plot->axisScaleDiv(i));
     }
   }
+  // Propagate this to spectrogram
+  if (spectrogram()) {
```
Comment (OwenArnold) [inline MantidPlot/src/Graph.cpp]: There's an isSpectrogram() method which would have been a more appropriate check that checking for a null pointer.

File: MantidPlot/src/Spectrogram.h [hunk 2]
```diff
@@ -220,6 +226,8 @@ class Spectrogram: public QObject, public QwtPlotSpectrogram, public MantidQt::A
   std::vector<unsigned char> mScaledValues;
   /// boolean flag to indicate intensity changed
   bool m_bIntensityChanged;
+  /// Flag for whether we autoscale color bar
+  bool d_color_map_autoscale;
```
Comment (OwenArnold) [inline MantidPlot/src/Spectrogram.h]: Coding standards for Mantid have all member variables pre-fixed with `m_` this is the only aspect of Hungarian format we follow.

**Comment by OwenArnold** [comment]
This is functionally correct. @tom-perkins please update release notes.

**Comment by tom-perkins** [comment]
[Release notes](http://www.mantidproject.org/ReleaseNotes_3_6_UI_Changes) have been updated.

**Comment by OwenArnold** [comment]
@tom-perkins  Could you please screenshot and highlight the new checkbox on the sliceviewer and add that to the release notes. As an additional issue, It would be great if you could also update the online documentation http://www.mantidproject.org/MantidPlot:_SliceViewer as it's now out of date.

**Comment by tom-perkins** [comment]
Added screenshot and updated documentation

**Pull request merged**